hibernate-core/src/main/java/org/hibernate/engine/spi/BatchFetchQueue.java

```diff
@@ -167,37 +171,36 @@ public void addBatchLoadableEntityKey(EntityKey key) {
 	public void removeBatchLoadableEntityKey(EntityKey key) {
 		if ( batchLoadableEntityKeys != null
 				&& key.isBatchLoadable( context.getSession().getLoadQueryInfluencers() ) ) {
-			final LinkedHashSet<EntityKey> set =  batchLoadableEntityKeys.get( key.getEntityName() );
+			final LinkedHashSet<EntityKey> set = NullnessUtil.castNonNull( batchLoadableEntityKeys ).get( key.getEntityName() );
```

Comment:
I know, it makes sense to NOT have that castNonNull, since this is inside a != null check, however CF doesn't seem to detect that one, and therefore complains of a potential null dereference ... :shrug:

Reply:
I figured out that if the nullcheck is last, there's no need for the nonnullcast ... which is in line with the side-effect theory.